portfolio/src/main/webapp/script.js

```diff
@@ -26,5 +26,28 @@ async function requestPuzzle() {
 
     //here it should be rendered to the page
     console.log(responseObject);
+    
+    const puzzlePage = document.getElementById("thisPuzzle");
+    const puzzleDifficulty = responseObject.difficulty;
+    const puzzleImage = responseObject.imageUrl;
+
+    if (puzzleDifficulty == "EASY"){
+        puzzlePage.innerHTML = `<img src=${puzzleImage} onLoad="snapfit.add((this), {level:1, mixed:true, snap:12, nokeys:true, aborder:true, polygon:true, aimage:true, aopacity: 0.3, space:15, callback:function(){solvedPuzzle()} });" width="750">`;
```

Comment:
I'd recommend refactoring this code to set the onLoad field programmatically instead of via innerHtml:
```suggestion
      // Note you'll have to add an image tag to viewpuzzle.html with the id 
      // "puzzle-image"
      const imageEl = document.getElementById("puzzle-image");
      imageEl.src = responseObject.imageUrl;
```

Reply:
Since the only thing that is changing in the object is the `level` field you can declare the rest of the fields separately and then set that specific field later:
```suggestion
   const settings = { 
      mixed: true, 
      snap: 12, 
      nokeys: true, 
      aborder: true, 
      polygon: true, 
      aimage: true, 
      aopacity: 0.3, 
      space: 15, 
      callback: solvedPuzzle
  };

  switch(responseObject.difficulty) {
      case "EASY":
          settings.level = 1;
          break;
      case "MEDIUM":
          settings.level = 3;
          break;
      case "HARD":
          settings.level = 5;
          break;
      default:
          console.warn("Unrecognized difficulty: " + responseObject.difficulty);
  }
  imageEl.onload = function() {
      snapfit.add(imageEl, settings);
  }
```